Sources/StreamChat/Controllers/ListDatabaseObserver_Tests.swift

```diff
@@ -163,6 +163,47 @@ class ListDatabaseObserver_Tests: XCTestCase {
         XCTAssertTrue(testFRC.test_performFetchCalled)
     }
     
+    func test_updateNotReported_whenSamePropertyAssignedViaAssignIfDifferent() throws {
+        // For this test, we need an actual NSFetchedResultsController, not the test one
+        let observer = ListDatabaseObserver<TestManagedObject, TestManagedObject>(
+            context: database.viewContext,
+            fetchRequest: fetchRequest,
+            itemCreator: { $0 }
+        )
+        
+        var receivedChanges: [ListChange<TestManagedObject>]?
+        observer.onChange = { receivedChanges = $0 }
+        
+        // Call startObserving to set everything up
+        try observer.startObserving()
+        
+        // Insert the test object
+        let testValue = String.unique
+        var item: TestManagedObject!
+        try database.writeSynchronously { _ in
+            let context = self.database.writableContext
+            item = NSEntityDescription.insertNewObject(forEntityName: "TestManagedObject", into: context) as? TestManagedObject
+            item.testId = testValue
+            item.testValue = testValue
+        }
+        
+        XCTAssertEqual(receivedChanges?.first?.item.testId, testValue)
+        
+        let oldChanges = receivedChanges
+        
+        // Assign the same testValue to the same entity
+        try database.writeSynchronously { _ in
+            // We use this function instead of direct assignment (`=`) in our `save` methods
+            assignIfDifferent(item, \.testValue, testValue)
+        }
+        
+        AssertAsync.staysEqual(receivedChanges, oldChanges)
+        
+        if case .update = receivedChanges?.first {
+            XCTFail("Update shouldn't be reported!")
+        }
```

Comment:
Isn't this assertion redundant when you already assert `staysEqual`?

Reply:
fixed in e1c35788df1afd10341996a90a43d21e32d4c39d